fdbserver/MasterServer.actor.cpp

```diff
@@ -982,7 +994,25 @@ ACTOR Future<Void> getVersion(Reference<MasterData> self, GetCommitVersionReques
 			}
 		}
 
-		rep.version = self->version;
+		Version repVersion = self->version;
+
+		if (req.splitID.present()) {
+			const auto& splitID = req.splitID.get();
+			bool inCache = splitIDVersionCache.exists(splitID) && splitIDPrevVersionCache.exists(splitID);
+
+			if (!inCache) {
+				splitIDVersionCache.add(splitID, repVersion);
+				splitIDPrevVersionCache.add(splitID, repPrevVersion);
+			} else {
+				// Do not advance self->version since the transaction is part of a previous versioned one.
+				self->version = repPrevVersion;
```

Comment:
this is potentially problematic -- basically you are rolling back versions, which is wrong. I don't think you need to do this.

For instance, let's say Split 1 get version 100. Master advances version to 120 and give to another transaction. Then Split 2 comes in, your change would roll back version to 100. Next transaction could get version 110, which is less than the already given version of 120.